src/Kernel/Context.class.st

```diff
@@ -1672,10 +1672,10 @@ Context >> runUntilErrorOrReturnFrom: aSender [
 	"Insert ensure and exception handler contexts under aSender"
 	error := nil.
 	context := aSender insertSender: (Context
-		contextOn: Error do: [:ex |
+		contextOn: Error, Halt do: [:ex |
 			error ifNil: [
 				"this is ugly but it fixes the side-effects of not sending an Unhandled error on Halt"
```

Comment:
if you remove the `isKindOf:`, this comment still has sense? it applies?

Reply:
Hmmm indeed, that comment could be removed.